tests/FakeItEasy.Specs/ArgumentMatchingSpecs.cs

```diff
@@ -0,0 +1,258 @@
+﻿namespace FakeItEasy.Specs
+{
+    using FluentAssertions;
+    using Xbehave;
+    using Xunit;
+
+    public static class ArgumentMatchingSpecs
+    {
+        public interface IFoo
+        {
+            int Bar(int a);
+
+            int Bar(int a, string b);
+
+            int Bar(int a, string b, bool c);
+
+            int Bar(int a, string b, bool c, object d);
+        }
+
+        [Scenario]
+        [InlineData(null, 0, 0)]
+        [InlineData(null, 0, 1)]
+        [InlineData(null, 0, 2)]
+        [InlineData(null, 0, 10)]

```

Comment:
I found the `0` confusing. Consider omitting and making `result` the last parameter?